controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -125,32 +129,49 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
 }
 
 func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
-
+	r.storageConsumer.Status.CephResources = []*ocsv1alpha1.CephResourcesSpec{}
 	if !r.storageConsumer.Spec.Enable {
-		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDisabled
-		return reconcile.Result{}, nil
+		if err := r.reconcileNonEnabledPhases(); err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		if err := r.reconcileEnabledPhases(); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
There is no need for the negative case to be first
```suggestion
	if r.storageConsumer.Spec.Enable {
		if err := r.reconcileEnabledPhases(); err != nil {
			return reconcile.Result{}, err
		}
	} else {
		if err := r.reconcileNonEnabledPhases(); err != nil {
			return reconcile.Result{}, err
		}
		```